src/hook/useStepper.ts

```diff
@@ -0,0 +1,171 @@
+import { List as ImmutableList, OrderedMap } from 'immutable'
+import type { Dispatch, Reducer } from 'react'
+import { useReducer } from 'react'
+import type { DeepReadonly, UseReducer } from 'superTypes'
+
+type StepStatus = 'disabled' | 'invalid' | 'completed' | 'uncompleted'
+
+export type StepperState = {
+  /**
+   * The id of the step in progress.
+   */
+  currentStepId: string | undefined
+  /**
+   * The status of each step.
+   */
+  stepsStatus: OrderedMap<string, StepStatus>
+  /**
+   * The list of non-disabled steps.
+   */
+  enabledSteps: ImmutableList<string>
```

Comment:
I understand why you don't write "The list of enabled steps", but still, it is confusing to have "non-disabled steps" in the description and `enabledSteps` as property name.
would it be a good idea to have "The list of enabled steps (invalid / completed / uncompleted) "
or something like that? or change the property name? 
or maybe I'm just nitpicking too much ;)

Reply:
You are right. I prefer to rename the property name to `nonDisabledSteps`.